openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/NoDataRasterSource.scala

```diff
@@ -0,0 +1,69 @@
+package org.openeo.geotrellis.layers
+
+import geotrellis.layer.CRSWorldExtent
+import geotrellis.proj4.{CRS, LatLng}
+import geotrellis.raster.io.geotiff.OverviewStrategy
+import geotrellis.raster.{UByteConstantNoDataCellType, UByteConstantTile, CellSize, CellType, GridBounds, GridExtent, MultibandTile, Raster, RasterMetadata, RasterSource, ResampleMethod, ResampleTarget, SourceName, TargetCellType, Tile, ubyteNODATA}
+import geotrellis.vector.Extent
+
+object NoDataRasterSource {
+  private val bandIndices: Seq[Int] = Seq(0)
+
+  def instance(gridExtent: GridExtent[Long], crs: CRS): NoDataRasterSource =
+    new NoDataRasterSource(UByteConstantNoDataCellType, gridExtent, crs)
+
+  def instance: NoDataRasterSource =
+    instance(GridExtent(LatLng.worldExtent, cols = 10, rows = 10), LatLng)
+}
+
+class NoDataRasterSource(override val cellType: CellType, override val gridExtent: GridExtent[Long], override val crs: CRS) extends RasterSource {
+  import NoDataRasterSource._
+
+  val targetCellType: Option[TargetCellType] = None
+
+  override def metadata: RasterMetadata = this
+
+  override protected def reprojection(targetCRS: CRS, resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): RasterSource =
+    new NoDataRasterSource(cellType, gridExtent.reproject(crs, targetCRS), targetCRS)
+
+  override def resample(resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): RasterSource =
+    new NoDataRasterSource(cellType, resampleTarget(gridExtent), crs)
+
+  override def read(extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]] = {
+    require(bands == bandIndices)
+
+    extent.intersection(gridExtent.extent)
+      .map { intersection =>
+        val intersectionGridBounds = gridExtent.gridBoundsFor(intersection).toGridType[Int]
+        val noDataTile = baseNoDataTile(intersectionGridBounds.width, intersectionGridBounds.height)
+        Raster(MultibandTile(noDataTile), intersection)
+      }
+  }
+
+  override def read(bounds: GridBounds[Long], bands: Seq[Int]): Option[Raster[MultibandTile]] = {
+    bounds.intersection(gridExtent.dimensions)
+      .map { intersection =>
+        val intersectionGridBounds = intersection.toGridType[Int]
+        val noDataTile = baseNoDataTile(intersectionGridBounds.width, intersectionGridBounds.height)
+        Raster(MultibandTile(noDataTile), gridExtent.extentFor(intersection))
+      }
+  }
+
+  private def baseNoDataTile(cols: Int, rows: Int): Tile = {
+    val noDataTile = UByteConstantTile(ubyteNODATA, cols, rows)
+    if (noDataTile.cellType == cellType) noDataTile else noDataTile.convert(cellType)
```

Comment:
Is this workaround still necessary?

Reply:
Doesn't look like it: `UByteConstantNoDataCellType` behaves better than `ByteConstantNoDataCellType`.